libraries/rush-lib/src/logic/buildCache/AdditionalFilesHasher.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { Import } from '@rushstack/node-core-library';
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+
+import type { IOptions } from 'glob';
+
+const glob: typeof import('glob') = Import.lazy('glob', require);
+
+const globAsync = (pattern: string, options: IOptions = {}): Promise<string[]> => {
+  return new Promise((resolve, reject) => {
+    glob(pattern, options, (err, matches) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      resolve(matches);
+    });
+  });
+};
+
+export class AdditionalFilesHasher {
+  public static async getFileHashedFilesMap(
+    globPatterns: string[],
+    cwd: string
+  ): Promise<Map<string, string>> {
+    const filePaths: Set<string> = await this._expandGlobPatterns(globPatterns, cwd);
+
+    return await this._calculateFileHashes(filePaths, cwd);
+  }
+
+  private static async _expandGlobPatterns(globPatterns: string[], cwd: string): Promise<Set<string>> {
+    const allMatches: Set<string> = new Set<string>();
+    for (const pattern of globPatterns) {
```

Comment:
Consider using `Async.forEachAsync` with a concurrency limit.

Reply:
https://github.com/microsoft/rushstack/pull/3824/commits/4f9fa41793daa125dcd48d290387fff659ae2bff

I've used it for both - glob expansion and reading files. 
I wasn't sure which concurrency level to pick, but I saw that `10` constant is used elsewhere in `rush-lib` so I've defaulted to that as well.

Reply:
Oh... and of course - I've tested it again with my rush projects, it seems that it is still working as expected.